api/evaluation/v1alpha1/common.go

```diff
@@ -77,3 +77,27 @@ type Parameter struct {
 type Report struct {
 	// TODO
 }
+
+// Define RAG common structure and variables
+const (
+	EvauateJobLabels = Group + "/rag"
+)
+
+func RagStatusChanged(a, b RAGStatus) bool {
+	if !a.CompletionTime.Equal(b.CompletionTime) {
+		return true
+	}
+	if a.Phase != b.Phase {
+		return true
+	}
+	ac, bc := a.Conditions, b.Conditions
+	la, lb := len(ac), len(bc)
+	if la != lb {
+		return true
+	}
+	if la == 0 {
+		return false
+	}
+	return ac[0].Type != bc[0].Type || ac[0].Status != bc[0].Status ||
```

Comment:
If we only keep one condition (ConditionReady I guess), we can use the `GetCondition(ConditionReady)` here

Reply:
There are two kinds of condition, `complete` and `failed`. This is used to indicate whether a job completes or fails at each stage.